File: Cpp/201 - 400/347 Top K Frequent Elements.cpp
// 8ms, 100.0%
static int x = []() {
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);
	return 0;
}();

class Solution
{
public:
	vector<int> topKFrequent(const vector<int> &nums, int k)
	{
		unordered_map<int, int> rec;
		for(auto n : nums) ++rec[n];
		vector<pair<int, int>> tmp(rec.begin(), rec.end());
		sort(tmp.begin(), tmp.end(), [](auto &x, auto &y) { return x.second >= y.second; });

		vector<int>ret;
		ret.reserve(k);
		for(int i = 0; i < k; ++i) ret.push_back(tmp[i].first);
		return ret;
	}
};

File: Python3/201 - 400/347 Top K Frequent Elements.py
# 44ms, 99.32%
class Solution:
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        from collections import Counter
        tmp = Counter(nums).most_common(k)
        return [x for x, _ in tmp]
